File: Data Structures & Algorithms/majority-element-ii/submission-0.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        List<Integer> ans = new ArrayList<>();

        Map<Integer, Integer> mp = new HashMap<>();
        for(int ele : nums){
            mp.put(ele, mp.getOrDefault(ele, 0)+1);
        }

        for(int key : mp.keySet()){
            int freq = mp.get(key);

            if(freq > nums.length/3){
                ans.add(key);
            }
        }

        return ans;
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums)-1

        while i <= j:
            mid = (i+j)//2
            m = nums[mid]

            if target > m:
                i = mid + 1
            elif target < m:
                j = mid - 1
            else:
                return mid
        return -1